provider-home/src/java/org/sakaiproject/archiver/provider/HomeArchiver.java

```diff
@@ -63,4 +77,70 @@ private String createHtmlFileContents(final String data) {
 		return StringEscapeUtils.unescapeHtml4(data);
 	}
 
+	/**
+	 * Archive images that are in the home frame and alter the home frame html to point to these locally saved images
+	 *
+	 * @param originalHtml
+	 * @param archiveId
+	 * @param siteId
+	 * @param toolId
+	 * @return updated html
+	 */
+	private String archiveImages(final String originalHtml, final String archiveId, final String siteId, final String toolId) {
+
+		final Document doc = Jsoup.parse(originalHtml);
+		final List<Element> imageElements = doc.select("img");
+
+		for (final Element e : imageElements) {
+			try {
+
+				// get the image bytes
+				final String imageSource = e.absUrl("src");
+				final byte[] bytes = Jsoup.connect(imageSource).ignoreContentType(true).execute().bodyAsBytes();
+
+				// get the file name
+				final String filename = getFileName(imageSource);
+
+				// set a maximum image width
+				e.attr("style", "max-width: 100%");
+				e.attr("height", "auto");
+
+				// archive the image
+				this.archiverService.archiveContent(archiveId, siteId, toolId, bytes, filename, "/images");
+
+				// change the src for this image in the html
+				e.attr("src", "images/" + filename);
+
+			} catch (final IOException e1) {
+				log.debug("Error when saving image from src: " + e.absUrl("src"));
+				continue;
+			}
+		}
+
+		return doc.html();
+	}
+
+	/**
+	 * Assign a filename for this image.
+	 *
+	 * If there is a file extension in the url, the filename will be the text from the last backslash to the file extension. Otherwise, a
+	 * random UUID will be assigned.
+	 *
+	 * @param imageSource url string
+	 * @return filename string
+	 */
+	private String getFileName(final String imageSource) {
+		final Pattern pattern = Pattern.compile(
```

Comment:
yes everything between the file extension and the backslash before it, i.e. 276.jpg would be extracted from http://getsokt.com/wp-content/uploads/2015/11/276.jpg